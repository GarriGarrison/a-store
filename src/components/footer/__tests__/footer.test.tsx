import { render, screen } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Footer } from '..'


describe('Тесты компонента Footer', () => {

  test('рендер текста - юридическое название фирмы', () => {
    render(
      <Router>
        <Footer />
      </Router>
    )

    const element = screen.getByText(/«Альфа Фьюче Пипл»/i)
    expect(element).toBeInTheDocument()
  })

  test('рендер текста - Политика конфиденциальности и обработки персональных данных', () => {
    render(
      <Router>
        <Footer />
      </Router>
    )

    const elementPolicy = screen.getByText(/Политика конфиденциальности/)
    expect(elementPolicy).toBeInTheDocument()

    const elementDataPersonal = screen.getByText(/и обработки персональных данных/)
    expect(elementDataPersonal).toBeInTheDocument()
  })

  test('снепшот компонента', () => {
    const { baseElement } = render(
      <Router>
        <Footer />
      </Router>
    )

    expect(baseElement).toMatchSnapshot()
  })
})
