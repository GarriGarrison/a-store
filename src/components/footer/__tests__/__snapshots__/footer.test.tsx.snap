// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Footer снепшот компонента 1`] = `
<body>
  <div>
    <footer
      class="container"
    >
      <div
        class="company"
      >
        © ООО «Альфа Фьюче Пипл», 2022
      </div>
      <div
        class="container"
      >
        <a
          class="link"
          href="/policy"
        >
          Политика конфиденциальности
          <br />
          и обработки персональных данных
        </a>
      </div>
    </footer>
  </div>
</body>
`;
